fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -8,6 +8,38 @@
 from fairlearn.metrics import DemographicParity, BoundedGroupLoss
```

Comment:
I'm guessing that the `metrics` versions of these can be removed? Everything should be handled as a moment.

Reply:
But first, we should cross-check that the new implementation works on all the tests for the old implementation! I didn't have a chance to do it and there isn't an easy one-to-one map--because of how the lagrange multipliers are represented between the two implementations.